lib/transaction.py

```diff
@@ -511,6 +511,36 @@ def is_schnorr_signed(self, input_idx):
                        for sig in self._inputs[input_idx].get('signatures', []))
         return False
 
+    def rpa_paycode_swap_dummy_for_destination(self,rpa_dummy_address, rpa_destination_address):
+        # This method was created for RPA - reusable payment address.  
+        # It swaps out a dummy destination address for the rpa-generated address.
+        # WARNING: This is not recommended for use outside of RPA since it could
+        # be dangerous to change the destination address of a transaction.
+ 
+        # This method expects cashaddr strings for parameters rpa_dummy_address and rpa_destination_address
+  
+        length = len (self._outputs)
+        i = 0
+        
+        while i < length:           
+            # Grab the address of each output...
+            loop_iteration_output = self._outputs[i]
+            loop_iteration_address = loop_iteration_output[1]
+             
+            # Compare the address to see if its the one we need to swap.  Note: 0 for the to_string = CASHADDR format
+            if loop_iteration_address.to_string(0) == rpa_dummy_address:
```

Comment:
Please don't use magic int literals.  `to_string(Address.FMT_CASHADDR)` for example.